packages/start/entry-client/mount.tsx

```diff
@@ -38,14 +38,18 @@ export default function mount(code?: () => JSX.Element, element?: Document) {
     mountRouter();
 
     async function mountIsland(el: HTMLElement) {
-      let Component = el.dataset.island && window._$HY.islandMap[el.dataset.island];
-      if (!Component || !el.dataset.hk) return;
-      _$DEBUG(
-        "hydrating island",
-        el.dataset.island,
-        el.dataset.hk.slice(0, el.dataset.hk.length - 1) + `1-`,
-        el
-      );
+      let componentLoader = el.dataset.componentId && window._$HY.islandMap[el.dataset.componentId];
+      if (!componentLoader || !el.dataset.hk) return;
+      const Component = (await componentLoader()).default;
+
+      if (import.meta.env.DEV) {
+        _$DEBUG(
+          "hydrating island",
+          el.dataset.island,
+          el.dataset.hk.slice(0, el.dataset.hk.length - 1) + `1-`,
+          el
+        );
+      }
```

Comment:
Because `window._$DEBUG` is only defined when `import.meta.env.DEV` is true (line 17), this was throwing an error.